packages/ButtonGroup/src/ButtonGroup.js

```diff
@@ -152,7 +177,11 @@ function ButtonGroup(props) {
 
   return (
     <sc.ButtonGroup {...props} ref={groupRef} onKeyDown={handleKeyDown}>
-      <ButtonGroupContext.Provider value={contextValue}>{children}</ButtonGroupContext.Provider>
+      <ButtonGroupContext.Provider value={contextValue}>
+        {React.Children.map(children, (element, index) => {
+          return React.cloneElement(element, { ref: childRefs[index] });
```

Comment:
I went to a fair bit of effort to avoid `cloneElement()` so this component would support children like this:

https://github.com/acl-services/paprika/blob/master/packages/ButtonGroup/stories/examples/Sandbox.js#L36-L47

(with wrapping fragments, `<divs>`, etc)

Reply:
I think you can probably avoid the need for a `childRefs[]` array, and instead utilize `focusButton()`

https://github.com/acl-services/paprika/blob/master/packages/ButtonGroup/src/ButtonGroup.js#L88-L91

Reply:
@mikrotron Unfortunately not from what I can tell, that `focusButton()` function is just calling focus on the Dom Element, not the react component that rendered it. So currently I'm finding creating dedicated Refs for each component rendered is the only way to achieve that...

Reply:
Ah, right, then it's focus highlight remains invisible... 😖 
I still think it's important to avoid `cloneElement()` though.  You may need to put the `childRefs[]` on `state`, then pass a setter on `context` to the items.  Then they can utilize a hook to set their refs and pass them back up.  I do something similar for `defaultIsActive`.